swaggen/src/main/java/utils/io/FileMapper.java

```diff
@@ -37,15 +47,79 @@
 	 * 
 	 * @param filename the name of the YAML file
-	 * @param object the Java object
+	 * @param newSwagger the Java object, typically swagger object 
+	 * @param <T> The type
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
+	 */
+	public static <T> void classToYaml(String filename, T newSwagger)
+			throws JsonGenerationException, JsonMappingException, IOException {
+		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+		File file = new File(filename);
+
+		checkExistingEndpoints(newSwagger, file, objectMapper);
+
+		if (newSwagger instanceof Swagger) {
+			Swagger toYaml = convertSwagger(newSwagger, filename, file, objectMapper);
+			objectMapper.writeValue(file, toYaml);
+		} else {
+			objectMapper.writeValue(file, newSwagger);
+		}
+	}
+
+	/**
+	 * Checks for existing endpoints in yaml files and if matched, add the existing
+	 * request methods to the new Swagger file.
+	 * @param swagger the swagger object containing all the endpoitns
+	 * @param file file object, containing where it is supposed to be placed
+	 * @param objectMapper the ObjectMapper
 	 * @throws JsonGenerationException
 	 * @throws JsonMappingException
 	 * @throws IOException
 	 */
-	public static <T> void classToYaml(String filename, T object)
-			throws JsonGenerationException, JsonMappingException, IOException {
-		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
-		objectMapper.writeValue(new File(filename), object);
+	private static <T> void checkExistingEndpoints(T swagger, File file, ObjectMapper objectMapper) 
+		throws JsonGenerationException, JsonMappingException, IOException {
+		if (file.exists() && swagger instanceof Swagger) {
+			Swagger existingSwagger = objectMapper.readValue(file, Swagger.class);
+			for (String newURL: ((Swagger)swagger).getPaths().keySet()) {
+				for (String oldURL: existingSwagger.getPaths().keySet()) {
+					if (newURL.equals(oldURL)) {
+						((Swagger)swagger).getPaths().get(newURL).putAll(existingSwagger.getPaths().get(oldURL));
+					}
+				}
+			}
+		}
 	}
 
+
+	/**
+	 * Finds the correct request method and endpoint within the swagger map given the filename, and writes
+	 * it to a yaml file
+	 * @param swagger the swagger object containing all the endpoints
+	 * @param filename name of the file
+	 * @param file file object, containing where it is supposed to be placed
+	 * @param objectMapper the ObjectMapper
+	 * @throws JsonGenerationException
+	 * @throws JsonMappingException
+	 * @throws IOException
+	 */
+	private static <T> Swagger convertSwagger(T swagger, String filename, File file, ObjectMapper objectMapper)
+		throws JsonGenerationException, JsonMappingException, IOException {
+		for (Map.Entry<String, Map<RequestMethod, Endpoint>> path: ((Swagger)swagger).getPaths().entrySet()) {
+			String url = path.getKey();
+			int basePathIndex = filename.lastIndexOf("/");

```

Comment:
@WilliamRGardiner is this because they are the same object? Or simply because there is no way of extracting the base path for each point and performing the comparison?

Reply:
Excuse me if I am incorrect here (I do not have the entire codebase here, so I am just reading the RP). It seems as though we are defining the basepath to be everything before the last instance of "/" in the path. We have let the user define the basepath and path elsewhere.

It appears to me that we are relying on these two definitions of basepath being identical. Is this assumption that we need both definitions to be identical correct? If so, this code is incorrect as a user may define a path with a "/" in it. Then part of the path would be assumed to be part of the basepath.